test/Microsoft.Azure.SignalR.IntegrationTests/MockService/MessageQueue.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR.IntegrationTests.MockService
+{
+    /// <summary>
+    /// Not a general purpose producer-consumer async queue implementation
+    /// Thread safe for only single concurrent reader + single concurrent writer
+    /// </summary>
+    internal class MessageQueue<T> where T : class
```

Comment:
reuse Channel?

Reply:
Great idea! We lost the Peek* capability but we didn't have a real need for it (`WaitToReadAsync` is enough so far).